src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs

```diff
@@ -225,7 +229,10 @@ private async Task<(Uri url, CancellationToken hostExitToken)> StartIISExpressAs
                     }
 
                     // Wait for the app to start
-                    if (!await started.Task)
+                    // The timeout here is large, because we don't know how long the test could need
+                    // We cover a lot of error cases above, but I want to make sure we eventually give up and don't hang the build
+                    // just in case we missed one -anurse
+                    if (!await started.Task.OrTimeout(TimeSpan.FromMinutes(10)))
```

Comment:
Consider replacing uses of `OrTimeout` with `TimeoutAfter`: https://github.com/aspnet/Testing/blob/dev/src/Microsoft.AspNetCore.Testing/TaskExtensions.cs